cql3/selection/selection.hh

```diff
@@ -172,6 +172,7 @@ private:
     std::unique_ptr<result_set> _result_set;
     std::unique_ptr<selectors> _selectors;
     const std::vector<size_t> _group_by_cell_indices; ///< Indices in \c current of cells holding GROUP BY values.
+    const std::optional<uint64_t> _limit; ///< Maximum number of rows to return.
```

Comment:
We may way well drop the optional and set the limit to numeric_limit::max() if it's  not specified.